merkle/log_verifier.go

```diff
@@ -248,3 +248,101 @@ func (v LogVerifier) VerifiedPrefixHashFromInclusionProof(
 	}
 	return res, nil
 }
+
+// VerifyRangeInclusion verifies that [begin, end) range, consisting of the
+// |leaves| with the specified hashes, is included into the tree with the
+// specified |size| and |root| hash, based on inclusion proofs for entries
+// number |begin| and |end-1| respectively.
+//
+// TODO(pavelkalinnikov): Current implementation has some level of redundancy,
+// as the combination of 2 inclusion proofs is a superset of the minimal range
+// inclusion proof. Consider adding proper range inclusion proof notion to
+// Trillian's API.
+func (v LogVerifier) VerifyRangeInclusion(
+	begin, end, size int64,
+	proof1, proof2 [][]byte,
+	root []byte, leaves [][]byte,
+) error {
+	if end <= begin {
+		return fmt.Errorf("empty range [%d,%d)", begin, end)
+	} else if got, want := int64(len(leaves)), end-begin; got != want {
+		return fmt.Errorf("wrong number of leaves %d, want %d", got, want)
+	}
+	if err := v.VerifyInclusionProof(begin, size, proof1, root, leaves[0]); err != nil {
+		return err
+	} else if err := v.VerifyInclusionProof(end-1, size, proof2, root, leaves[len(leaves)-1]); err != nil {
```

Comment:
this `else` can go too (I think it's less dense and easier to read without, do you agree?)